src/state/store.ts

```diff
@@ -0,0 +1,136 @@
+import { IFieldContext, IField, FieldRegistry } from '../data';
+
+export interface ISubscriber {
+    (): void
+}
+
+export interface FieldEditingHandler {
+    (field: IField): void;
+}
+
+// createStore is a helper function for initializing
+// the store.
+export function createStore(props: IStoreProps) {
+    return new Store(props);
+}
+
+export interface IStore {
```

Comment:
At beginning, I am considering only put editingContext inside in a bridge or even just a message channel between formbuilder and optionEditor.